scipy/interpolate/_rgi_cython.pyx

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+cimport numpy as np
+
+import itertools
+
+
+def evaluate_linear(values, indices, norm_distances, out_of_bounds):
+    # slice for broadcasting over trailing dimensions in self.values
+    vslice = (slice(None),) + (None,) * (values.ndim - len(indices))
+
+    # Compute shifting up front before zipping everything together
+    shift_norm_distances = [1 - yi for yi in norm_distances]
+    shift_indices = [i + 1 for i in indices]
+
+    # The formula for linear interpolation in 2d takes the form:
+    # values = self.values[(i0, i1)] * (1 - y0) * (1 - y1) + \
+    #          self.values[(i0, i1 + 1)] * (1 - y0) * y1 + \
+    #          self.values[(i0 + 1, i1)] * y0 * (1 - y1) + \
+    #          self.values[(i0 + 1, i1 + 1)] * y0 * y1
+    # We pair i with 1 - yi (zipped1) and i + 1 with yi (zipped2)
+    zipped1 = zip(indices, shift_norm_distances)
+    zipped2 = zip(shift_indices, norm_distances)
+
+    # Take all products of zipped1 and zipped2 and iterate over them
+    # to get the terms in the above formula. This corresponds to iterating
+    # over the vertices of a hypercube.
+    hypercube = itertools.product(*zip(zipped1, zipped2))
+    value = np.array([0.])
+    for h in hypercube:
+        edge_indices, weights = zip(*h)
+        weight = np.array([1.])
+        for w in weights:
+            weight = weight * w
+        value = value + np.asarray(values[edge_indices]) * weight[vslice]
+    return value
+
+
+def find_indices(grids, bounds_error, xi):
+    # find relevant edges between which xi are situated
+    indices = []
+    # compute distance to lower edge in unity units
+    norm_distances = []
+    # check for out of bounds xi
+    out_of_bounds = np.zeros((xi.shape[1]), dtype=bool)
+    # iterate through dimensions
+    for x, grid in zip(xi, grids):
+        i = np.searchsorted(grid, x) - 1
```

Comment:
https://github.com/scipy/scipy/blob/main/scipy/interpolate/_ppoly.pyx#L570 for a cdef version of searchsorted. It's a scalar version, so likely needs a loop of the components of `i`